### ISSUE TITLE
[Proposal] Support Office accounts without Administrator approval

### ISSUE THREAD

**Issue by clragon** [bkaankose/Wino-Mail#268]
**Describe the solution you'd like**
When adding Office accounts, the sign in requires the administrator of your Office instance to specifically allow this App.
This is essentially impossible to pull off on University E-Mails.
I dont know how this technically works, but it should be possible to use Office accounts without this requirement, as seen in many other E-Mail clients.

**Is your proposal implemented in Windows Mail?**
Yes. Maybe. Windows Mail does not require an Administrator to approve this App.
That might be because its a Microsoft app and because its always allowed by default or because it uses some other method of accessing the accounts, though I assume its the former.

**Additional context**
This is a major roadblock and prevents me from migrating my entire E-Mail usage to this app. 
The administrators of my University and Workplace do not care what E-Mail app I wish to use and they will not grant me access.

**Comment by Tiktack** [comment]
Can you give an example of mail client which doesn't ask about approval and it's not MS app?

**Comment by clragon** [comment]
sure. I am currently using [eM client](https://www.emclient.com/) which does not require administrator approval for me to use my office accounts.

**Comment by bkaankose** [comment]
eM client also requires permission for my work account.

![image](https://github.com/user-attachments/assets/b61c5d74-7118-4dac-8dd7-97749f7b14f5)


It really depends how strict your administrator is. So far I only found out [Owl for Exchange](https://addons.thunderbird.net/en-us/thunderbird/addon/owl-for-exchange/) plugin works well with Thunderbird. They are using Outlook Web Access (OWA) to handle things in the background probably but this is something that requires high amount of work to integrate it in Wino.

Maaaybe maybe maybe... It's possible with using customized WebView2 as a synchronizer but in order for this strategy to work Wino's core synchronization mechanism must change a little.

Basically app's shell will work as a standalone UI app and all synchronizations must be handled in the background via separate proces, let's call it server. Server delivers changes to client (App) and it may work, but as I said, tons of work.

**Comment by Tiktack** [comment]
The same as for @bkaankose. I have two orgs, one requires only my approval, but second org requires Admin Approval. 
![image](https://github.com/user-attachments/assets/ad929c33-f290-4084-b7d2-70e6cf3106b7)

**Comment by bkaankose** [comment]
Keep in mind, your IT department can detect if you are using OWA or not. So if they are super uber duper strict, you might get multiple re-authentication requests with that plugin as well.

**Comment by clragon** [comment]
oh, I see. that seems like a very unfortunate situation. what exactly does it mean to be "verified"? 
I assume that might not be possible for Wino?

**Comment by Tiktack** [comment]
Verified this is the blue badge that you can see after "eM Client s.r.o.". But getting this badge not that simple, especially for open source project. Here is a MS doc about that. 
https://learn.microsoft.com/en-us/entra/identity-platform/publisher-verification-overview

And here is example what it does, even my first account for which in eM client I was able to give approve, don't give me this chance
![image](https://github.com/user-attachments/assets/fee7a55e-bb08-4601-911e-1c32f6cbf45e)

**Comment by TonyBnz** [comment]
I also have the problem shown above by Tiktack and am wondering if running Wino with administer privileges would get round it, unfortunately there doesn't seem to be a way to do this.

I do volunteer work for an organisation that supports the public and so need to set up their emails on whichever client I use on my desktop.  I have successfully set up the account in Outlook, Mailbird, Bluemail,  Edison and the Bat without any problems.  Why so many?  I am looking for an alternative to Outlook and Wino would be my preference if I can get round the above problem.

**Comment by EmirLogas** [comment]
![image](https://github.com/user-attachments/assets/211a037f-6afe-498f-8e0c-a5ce49bcbd46)

Same Issue for me. There was no such problem in different mail applications. If this error can be solved, I want to try Wino Mail.

**Comment by 1reeper** [comment]
Same issue as OP, university email cannot be added to Wino.  Mailspring allowed the same email address by prompting a login through browser without any admin access issues.  I prefer the UI and usability of Wino mail so I'm hoping this issue can be addressed.

**Comment by pfalkingham** [comment]
> Same issue as OP, university email cannot be added to Wino. Mailspring allowed the same email address by prompting a login through browser without any admin access issues. I prefer the UI and usability of Wino mail so I'm hoping this issue can be addressed.

Unfortunately Mailspring doesn't do anything special - even going through the web I'm then met with an approval request page.  OWL for exchange is the only thing outside Outlook and Apple mail that has worked, and I think that sends a client-ID pretending to be outlook.

**Comment by Abdullah-Haque-Selise** [comment]
Can you please add it to  Azure App Gallery so that we can add it via entra ID enterprise app settings.

**Comment by dsaderholm** [comment]
I +1 this as well. I'm stuck in the same situation with my college account — it only works with:

- OWL for Exchange (Thunderbird plugin)
- Windows Mail & Calendar (although, RIP 🪦 — Microsoft account syncing officially broke on April 11, 2025)
- Nine (Android app, I have no idea why)
- and of course, all current Outlook apps (desktop, web, and mobile).

That's it. Nothing else works.
I even tried all the methods in DavMail, including the newer "O365Interactive" option — none of them were successful either.
It seems like only a handful of apps (like OWL and Nine) figured out how to do true OWA-based scraping without needing admin approval.

Really hoping Wino can eventually support something like this!

**Comment by btdat2506** [comment]
Same issue here. No problem with Thunderbird, but can't get it working here.

![Image](https://github.com/user-attachments/assets/34ed3ed5-c143-4eb5-8837-0e265a0923bd)

**Comment by pfalkingham** [comment]
I've been looking into this a bit more, and it might not be solvable by WinoMail.  The reason Thunderbird with OWL works is that Thunderbird is built on firefox, so appears as a browser to outlook servers.

If you just use an outlook clientID with an app, including something like DavMail, the redirect URI doesn't match, and it doesn't work without admin adding the app to their azure dashboard.

Owl and Thunderbird don't pretend, they literally _are_ a browser accessing outlook via the web.

In those cases, anything that works has been added by Admin.  I don't think there's a way around this.

At least, that's my understanding after more digging.
